File: CPlus/Pow.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double result=1.0;
        for (int i=n;i!=0;i=i/2){
            if (i%2!=0) result*=x;
            x*=x;
        }
        return n < 0 ? 1 / result : result;
    }
};

File: Python/pow.py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        negative=False
        if n<0:
            negative=True
            n=abs(n)
        if n==0:
            return 1
        if x==0:
            return 0
        
        ans=1.0
        cur_ans=x
        
        while n>0:
            if n%2==1:
                ans=ans* cur_ans
            
            cur_ans*=cur_ans
            n=n//2
        
        if negative:
            return 1.0/ans
        else:
            return ans